src/System.Net.Http/src/System/Net/Http/Unix/CurlTransportContext.cs

```diff
@@ -2,33 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Security.Authentication.ExtendedProtection;
+using System.Threading;
 
 namespace System.Net.Http
 {
-    internal class CurlTransportContext : TransportContext
+    internal sealed class CurlTransportContext : TransportContext
     {
-        private CurlChannelBinding _channelBinding = new CurlChannelBinding();
+        private CurlChannelBinding _channelBinding;
 
         internal CurlChannelBinding CurlChannelBinding
         {
             get
             {
-                return _channelBinding;
+                CurlChannelBinding binding = Volatile.Read(ref _channelBinding);
+                if (binding == null)
+                {
+                    var newBinding = new CurlChannelBinding();
+                    binding = Interlocked.CompareExchange(ref _channelBinding, newBinding, null);
```

Comment:
I assume you're doing all this Interlocked stuff because you are trying to protect against multi-threaded access to the `GetChannelBinding` method?  In practice, that almost never happens but in theory it could.  The method is really only useful to be called while preparing to SerializetoStremAsync() a custom HttpContent that is being posted.  And the CBT (Channel Binding Token) needs to be retrieved to validate the security of the channel.

On a side note, this is an interesting coding pattern for lazy, mulit-thread aware, initialization.  Instead of a lock(object), you are getting the equivalent with Interlocked.CompareExchange.